internal/binder/function/funcs_obj.go

```diff
@@ -155,4 +156,51 @@ func registerObjectFunc() {
 		},
 		check: returnNilIfHasAnyNil,
 	}
+	builtins["erase"] = builtinFunc{
+		fType: ast.FuncTypeScalar,
+		exec: func(ctx api.FunctionContext, args []interface{}) (interface{}, bool) {
+			contains := func(array []string, target string) bool {
+				for _, v := range array {
+					if target == v {
+						return true
+					}
+				}
+				return false
+			}
+			if len(args) != 2 {
+				return fmt.Errorf("the argument number should be 2, got %v", len(args)), false
+			}
+			res := make(map[string]interface{})
+			argMap, ok := args[0].(map[string]interface{})
+			if !ok {
+				return fmt.Errorf("the first argument should be map[string]interface{}, got %v", args[0]), false
+			}
+			eraseArray := make([]string, 0)
+			v := reflect.ValueOf(args[1])
+			switch v.Kind() {
+			case reflect.Slice, reflect.Array:
+				array := args[1].([]string)
```

Comment:
The args[1] is likely to be []any, the type cast here will fail. You can use cast.ToStringSlice here to do the conversion.

Reply:
Thanks! Fixed.

Reply:
ah, here I mean "github.com/lf-edge/ekuiper/pkg/cast"

Reply:
> ah, here I mean "github.com/lf-edge/ekuiper/pkg/cast"

Thanks the `good first issue` 😅